src/js/components/modals/ServiceFormModal.js

```diff
@@ -329,6 +338,15 @@ class ServiceFormModal extends mixin(StoreMixin) {
         this.props.isEdit,
         JSON.parse(this.state.service.toJSON())
       );
+
+      // Handle service rename if this action renamed a field
+      if (this.props.isEdit && this.sourceService) {
+        let sourceName = this.sourceService['id'].split('/').pop();
```

Comment:
Can't we simplify this to use `this.props.service.getName()`? Let's not replicate the `getName` functionality.